google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -189,17 +190,21 @@ def __del__(self):
         """
         logger.debug("Entering deconstructor")
 
-        if self._current is not None:
-            logger.debug("Waiting for _current_instance_data to finish")
-            self._current.cancel()
-
-        if not self._client_session.closed:
-            logger.debug("Waiting for _client_session to close")
-            close_future = asyncio.run_coroutine_threadsafe(
-                self.__client_session.close(), loop=self._loop
-            )
-            close_future.result()
-
+        async def _deconstruct():
+            if not self._client_session.closed:
+                logger.debug("Waiting for _client_session to close")
+                await self._client_session.close()
+            if self._current is not None:
+                logger.debug("Waiting for _current to be cancelled")
+                await complete_task_cancellation(self._current)
+            if self._next is not None:
+                logger.debug("Waiting for _next to be cancelled")
+                await complete_task_cancellation(self._next)
```

Comment:
It's probably better to use `asyncio.gather` to await all three, so they can be actually async

Reply:
done